src/os/windows/crashlog_win.cpp

```diff
@@ -462,44 +462,51 @@ static INT_PTR CALLBACK CrashDialogFunc(HWND wnd, UINT msg, WPARAM wParam, LPARA
 {
 	switch (msg) {
 		case WM_INITDIALOG: {
+			size_t crashlog_length = CrashLogWindows::current->crashlog.size() + 1;
+			/* Reserve extra space for LF to CRLF conversion. */
+			crashlog_length += std::count(CrashLogWindows::current->crashlog.begin(), CrashLogWindows::current->crashlog.end(), '\n');
+
+			const size_t filename_count = 4;
+			const size_t filename_buf_length = MAX_PATH + 1;
+			const size_t crash_desc_buf_length = lengthof(_crash_desc) + filename_buf_length * filename_count + 1;
+
 			/* We need to put the crash-log in a separate buffer because the default
-			 * buffer in MB_TO_WIDE is not large enough (512 chars) */
-			wchar_t filenamebuf[MAX_PATH * 2];
-			wchar_t crash_msgW[65536];
-			/* Convert unix -> dos newlines because the edit box only supports that properly :( */
-			const char *unix_nl = CrashLogWindows::current->crashlog.data();
-			char dos_nl[65536];
-			char *p = dos_nl;
+			 * buffer in MB_TO_WIDE is not large enough (512 chars).
+			 * Use VirtualAlloc to allocate pages for the buffer to avoid overflowing the stack.
+			 * Avoid the heap in case the crash is because the heap became corrupted. */
+			const size_t total_length = crash_desc_buf_length * sizeof(wchar_t) + \
+										crashlog_length * sizeof(wchar_t) + \
+										filename_buf_length * sizeof(wchar_t) * filename_count + \
+										crashlog_length;
+			void *raw_buffer = VirtualAlloc(nullptr, total_length, MEM_COMMIT | MEM_RESERVE, PAGE_READWRITE);
+
+			wchar_t *crash_desc_buf = reinterpret_cast<wchar_t *>(raw_buffer);
+			wchar_t *crashlog_buf = crash_desc_buf + crash_desc_buf_length;
+			wchar_t *filename_buf = crashlog_buf + crashlog_length;
+			char *crashlog_dos_nl = reinterpret_cast<char *>(filename_buf + filename_buf_length * filename_count);
+
+			/* Convert unix -> dos newlines because the edit box only supports that properly. */
+			const char *crashlog_unix_nl = CrashLogWindows::current->crashlog.data();
+			char *p = crashlog_dos_nl;
 			char32_t c;
-			while ((c = Utf8Consume(&unix_nl)) && p < lastof(dos_nl) - 4) { // 4 is max number of bytes per character
+			while ((c = Utf8Consume(&crashlog_unix_nl))) {
```

Comment:
The removal of the `p <` check is safe, as `crashlog_dos_nl` is calculated to be at least as wide as `crashlog_unix_nl`. So it is not possible for the latter to return a character while the first can't fit it.

Reply:
And `\n` are counted twice to make room for `\r` in `crashlog_dos_nl`. I think I can confirm your claim :).